swingset/src/main/java/com/nqadmin/swingset/SSBaseComboBox.java

```diff
@@ -332,6 +401,144 @@ public boolean hasSelection() {
 		Object item = getSelectedItem();
 		return item != null && item != nullItem;
 	}
+	
```

Comment:
I rewrote/reordered things here and reworked the comments. I think the description aligns with the code. Note that I have not encountered SCENARIO 1-D or SCENARIO 3 in testing, but felt safer providing some code to handle. The user would get a popup if encountered so we should hear about it.